src/mbgl/map/geometry_tile.hpp

```diff
@@ -30,6 +31,7 @@ class GeometryTileFeature : private util::noncopyable {
     virtual ~GeometryTileFeature() = default;
     virtual FeatureType getType() const = 0;
     virtual mapbox::util::optional<Value> getValue(const std::string& key) const = 0;
+    virtual std::unordered_map<std::string, std::string> getAllValues() const = 0;
```

Comment:
Yes, because in building the abstract API for all `GeometryTile` classes, we have to support `VectorTile`, which creates its map from raw vector tile data. Applying a patch such as https://gist.github.com/incanus/d4302ba58616c5096442 to return references doesn't work because we need to create the map on the stack and get: 

```
.../src/mbgl/map/vector_tile.cpp:99:12: Reference to stack memory associated with local variable 'values' returned
```